subworkflows/nf-core/bam_variant_calling_haplotypecaller/main.nf

```diff
@@ -0,0 +1,114 @@
+include { GATK4_HAPLOTYPECALLER  } from '../../../modules/nf-core/gatk4/haplotypecaller/main'
+include { BEDTOOLS_SPLIT         } from '../../../modules/nf-core/bedtools/split/main'
+include { BCFTOOLS_MERGE         } from '../../../modules/nf-core/bcftools/merge/main'
+include { TABIX_TABIX            } from '../../../modules/nf-core/tabix/tabix/main'
+
+workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
+
+    take:
+    ch_bam              // channel: [mandatory] [meta, bam, bai, bed]
+    ch_fasta            // channel: [mandatory] [fasta]
+    ch_fasta_fai        // channel: [mandatory] [fasta_fai]
+    ch_dict             // channel: [mandatory] [dict]
+    ch_dragstr_models   // channel: [optional]  [meta, dragstr_model]
+    ch_dbsnp            // channel: [optional]  [dbsnp]
+    ch_dbsnp_tbi        // channel: [optional]  [dbsnp_tbi]
+
+    strategy            // string:  [mandatory] => The strategy used to run the subworkflow ["scatter","no_scatter"]
+    scatter_count       // integer: [optional]  => The amount of scattered BEDs should be created
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    //
+    // Optional: Scatter the BED files
+    //
+    if (strategy == "scatter" && scatter_count > 1){
+
+        ch_beds = ch_bam.map({ meta, bam, bai, bed -> [ meta, bed ]})
+
+        BEDTOOLS_SPLIT (
+            ch_beds,
+            scatter_count
+        )
+        ch_versions = ch_versions.mix(BEDTOOLS_SPLIT.out.versions.first())
+
+        ch_scattered_bams = ch_bam.combine(BEDTOOLS_SPLIT.out.beds.transpose(), by:0)
+                                .map({ meta, bam, bai, full_bed, scattered_bed ->
+                                    new_meta = meta.clone()
+                                    new_meta.id = scattered_bed.baseName
+                                    new_meta.sample = meta.id
+                                    [ new_meta, bam, bai, scattered_bed ]
+                                })
+
+    } else {
+        ch_scattered_bams = ch_bam
+    }
```

Comment:
So for each variantcalling subworkflow we have to resplit the bedfile?

Reply:
and you need to know the number of splits you want to go for? Does the tool evenly split the bedfile into regions of similar size?

Reply:
> So for each variantcalling subworkflow we have to resplit the bedfile?

Yeah or we could maybe adjust this workflow to also include multiple variant callers? (a bit similar to the uber alignment subworkflow)

> and you need to know the number of splits you want to go for? Does the tool evenly split the bedfile into regions of similar size?

bedtools split has two options for its splitting algorithm. The default uses a split to create regions that are similar in size with the downside that you could split on a variant region. The other algorithm keeps the regions specified in the BED file and tries to split them evenly (but there will be some variation in region length with this algorithm)